### PR TITLE
New Ruin: Singularity Research Lab

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request

Adds the Singularity Research Lab, formerly a cutting edge science station, now overrun with kudzu, it is a space ruin.

The two areas of note are the singularity reactor, which is assembled, and would just need a hand if someone were to want to start it, and the research lab. The Research lab contains the fruits of the now deceased science staff's labors, assorted energy weapons. Unfortunately, it also contains the deceased science staff.
dreamseeker_HFLqhdKLV5
Other areas of the base were not so lucky, and are thoroughly infested
image
The engineering team attempted to hold back the vines, and quickly discovered that fire was not sufficient.
dreamseeker_IrJikGDXKw
And what used to be the recreation area is completely gone

Notably, the hangar is empty. I plan on making a patch to put a subshuttle inside it once that rolls around.

Notable loot includes:
3 energy SMGs
3 Flamethrowers
The Ion Projector, a self charging Ion weapon.
An Antique Laser
2 Energy PDWs
2 Accelerator Laser Cannons
4 engineering hardsuits
An engineering lathe and circuit imprinter
A particle accelerator
A singularity generator
6 emitters
1 energy shotgun
Kudzu Seeds
Basically Everything You'd Need For an R&D Set Up
A sense of pride and accomplishment

I feel like this has some rough spots but I've got no idea where to start, so into the review -> testing -> feedback process it goes

- [x] The ruin spawns when the spawn ruin verb doesn't runtime.
## Why It's Good For The Game
More ruin variety. This one spawns in space and does a few things that I haven't seen yet. Mainly a singularity, cool semi-hidden asteroid base that could in theory, be turned into a player lair.

## Changelog

:cl:
add: An abandoned Nanotrasen Asteroid Facility has been spotted in the area. Salvage teams are advised to steer clear, or at least bring a knife.
add: kudzu zombie subtype. 
fix: vent iconstates.
/:cl:

### CODE REVIEW

File: _maps/map_catalogue.txt [hunk 1]
```diff
@@ -462,6 +462,9 @@ Find the key for using this catalogue in "map_catalogue_key.txt"
 		Size = (x = 19)(y = 15)(z = 1)
 		Tags = "No Combat", "Minor Loot", "Shelter"
 
+		File Name = "_maps\RandomRuins\SpaceRuins\singularitylab.dmm"
+		Size = (x = 121)(y = 75)(z = 1)
+		Tags = "Boss Combat Challenge", "Major Loot", "Bad Shelter", "Shelter"
```
Comment (Latentish) [inline _maps/map_catalogue.txt]: Remove "Bad Shelter". Bad shelter would be like if you have a building with no air on a jungle planet. Reverse of shelter.

File: code/game/area/areas/ruins/space.dm [hunk 2]
```diff
@@ -549,3 +549,33 @@
 	name = "Syndicate Laboratory 4071 Surrounding Area"
 	icon_state = "dk_yellow"
 	color = "#26773a88"
+
+//Singularity Lab
```
Comment (Latentish) [inline code/game/area/areas/ruins/space.dm:533]: Try to cut down on areas.

**Comment by Latentish** [comment]
I like ruin. I don't have much to say about it really. No complaints other than needing to cut down on areas and fixing the catalogue entry.

**Comment by Apogee-dev** [comment]
Probably needs some indestructible walls around some loot areas to minimize cheesing with things like jackhammers, plasma cutters, and mechs

**Review by ZephyrTFA — APPROVED** [review]

**Review by Latentish — APPROVED** [review]
I've been waiting for Bjarl to decide they were done with this, and Apogee suggested indestructible areas, but having observed multiple people run this I can say that busting into different places on this is satisfactory and still pretty deadly overall. So I'm going to approve this anyway.
I'd like to merge this, if it's not ready I'd ask you to close or draft it.

**Comment by Erikafox** [comment]
It has consistently reproducible issues with proper spawning that I'm hoping tmt can give more insight on.

**Comment by Latentish** [comment]
MERGEMA WHAT IS MERGEMA

**Review by spockye — CHANGES_REQUESTED** [review]
replace the space cash with the new money, and it's merging'o'clock

**Review by spockye — APPROVED** [review]
